adapters/sharethrough/sharethrough.go

```diff
@@ -130,20 +130,36 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 
 	bidderResponse := adapters.NewBidderResponse()
 	bidderResponse.Currency = "USD"
+	var errors []error
 
 	for _, seatBid := range bidResp.SeatBid {
-		for i := range seatBid.Bid {
-			bidType := openrtb_ext.BidTypeBanner
-			if bidReq.Imp[0].Video != nil {
-				bidType = openrtb_ext.BidTypeVideo
+		for _, bid := range seatBid.Bid {
+			bidType, err := getMediaTypeForBid(bid)
+			if err != nil {
+				errors = append(errors, err)
 			}
 
 			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
 				BidType: bidType,
-				Bid:     &seatBid.Bid[i],
+				Bid:     &bid,
```

Comment:
Please use `&seatBid.Bid[i]` as you were doing previously. With this updated way, you will end up with a pointer pointing at the LAST element in the seatBid.Bid slice (since the iteration stops at that point) because bid variable is getting a copy of each entry, not the entry itself.

Reply:
Thanks I changed it here: https://github.com/prebid/prebid-server/pull/2780/commits/22f2e3e9de0b7753ab962814c5fbb9e4e25d3714